CMakeLists.txt

```
@@ -7,15 +7,34 @@ option(QHOTKEY_INSTALL "Enable install rule" ON)
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
-find_package(Qt5 COMPONENTS Core Widgets REQUIRED)
+set(MINIMUM_REQUIRED_QT_MAJOR 5)
+set(MINIMUM_REQUIRED_QT6 6.2.0)
+set(QT_MAJOR ${MINIMUM_REQUIRED_QT_MAJOR} CACHE STRING "Qt major version to use")
+
+if(QT_MAJOR GREATER_EQUAL ${MINIMUM_REQUIRED_QT_MAJOR})
+    message(STATUS "Info: building with Qt${QT_MAJOR}")
+    if(QT_MAJOR EQUAL 6)
+        find_package(Qt${QT_MAJOR} ${MINIMUM_REQUIRED_QT6} COMPONENTS Core Widgets REQUIRED)
+    else()
+        find_package(Qt${QT_MAJOR} COMPONENTS Core Widgets REQUIRED)
+    endif()
+else()
+    message(FATAL_ERROR "Qt major version must be at least ${MINIMUM_REQUIRED_QT_MAJOR}")
+endif()
```

I would like to simplify the logic a bit to avoid custom messages, let CMake be responsible for that:

```suggestion
set(QT_MAJOR 5 CACHE STRING "Qt major version to use")

if(QT_MAJOR EQUAL 6)
    find_package(Qt${QT_MAJOR} 6.2.0 COMPONENTS Core Widgets REQUIRED)
else()
    find_package(Qt${QT_MAJOR} COMPONENTS Core Widgets REQUIRED)
endif()
```

---

Nice.

Instead of using `if(QT_MAJOR EQUAL 6)`, I think it's better to use `if(QT_MAJOR GREATER_EQUAL 6)`, as this will favor the long run. I used `GREATER_EQUAL` in lower lines for dealing with X11/X11Extras CMake packages. Agree?

---

I would leave `EQUAL` here because for the situation with theoretical Qt 7 it is better to go into the else block since it doesn't have a hardcoded 6.2.0 requirement. What do you think?

---

I see. The X11/X11Extras logic in lower lines is different, since it really applies to Qt 6 or higher where there is no X11Extras.

Committing.